strapi/strapi
# Settings: Re-fetch users list every time

### What does it do?

Removes caching from the settings user-list.

### Why is it needed?

Currently, when a user is updated, the list is not re-fetched. We could either invalidate the query after a user has been edited, or re-fetch every time. I find the latter less opinionated and more in-line with the rest of the app.

> **Note**
> The edit view of user is terribly broken (roles are removed when re-saving a user, fields are not updated ...). There is an open issue (https://github.com/strapi/strapi/issues/15673) and we have this in the content-squad backlog.

### How to test it?

1. Navigate to http://localhost:1337/admin/settings/users?pageSize=10&page=1&sort=firstname
2. Edit a user
3. Go  back to list
4. Verify the user has been updated there too
